components/apimgt/org.wso2.carbon.apimgt.rest.api.endpoint.registry/src/main/java/org/wso2/carbon/apimgt/rest/api/endpoint/registry/impl/RegistriesApiServiceImpl.java

```diff
@@ -63,24 +65,34 @@ public Response getAllEntriesInRegistry(String registryId, MessageContext messag
 
     @Override
     public Response getRegistryByUUID(String registryId, MessageContext messageContext) {
-        RegistryDTO registry = new RegistryDTO();
-        registry.setId("01234567-0123-0123-0123-012345678901");
-        registry.setMode(RegistryDTO.ModeEnum.WRITE);
-        registry.setType(RegistryDTO.TypeEnum.WSO2);
-        registry.setName("Dev Registry");
-        return Response.ok().entity(registry).build();
+        RegistryDTO registryDTO = null;
+        EndpointRegistry registryProvider = new EndpointRegistryImpl();
+        try {
+            EndpointRegistryInfo endpointRegistryInfo = registryProvider.getEndpointRegistryByUUID(registryId);
+            if (endpointRegistryInfo != null) {
```

Comment:
Shall we send a 404 when the registry cannot be found for the given uuid?

Reply:
i think currently it will return a null with 200 http code.

Reply:
Fixed